iris-mpc-store/src/s3_importer.rs

```diff
@@ -0,0 +1,259 @@
+use crate::StoredIris;
+use async_trait::async_trait;
+use aws_sdk_s3::Client;
+use bytes::Bytes;
+use futures::{stream, Stream, StreamExt};
+use iris_mpc_common::{IRIS_CODE_LENGTH, MASK_CODE_LENGTH};
+use rayon::{iter::ParallelIterator, prelude::ParallelBridge};
+use serde::Deserialize;
+use std::{io::Cursor, mem, pin::Pin};
+use tokio::task;
+
+const SINGLE_ELEMENT_SIZE: usize = IRIS_CODE_LENGTH * mem::size_of::<u16>() * 2
+    + MASK_CODE_LENGTH * mem::size_of::<u16>() * 2
+    + mem::size_of::<u32>(); // 75 KB
+const CSV_BUFFER_CAPACITY: usize = SINGLE_ELEMENT_SIZE * 10;
+
+#[async_trait]
+pub trait ObjectStore: Send + Sync + 'static {
+    async fn get_object(&self, key: &str) -> eyre::Result<Bytes>;
+    async fn list_objects(&self) -> eyre::Result<Vec<String>>;
+}
+
+pub struct S3Store {
+    client: Client,
+    bucket: String,
+}
+
+impl S3Store {
+    pub fn new(client: Client, bucket: String) -> Self {
+        Self { client, bucket }
+    }
+}
+
+#[async_trait]
+impl ObjectStore for S3Store {
+    async fn get_object(&self, key: &str) -> eyre::Result<Bytes> {
+        let result = self
+            .client
+            .get_object()
+            .bucket(&self.bucket)
+            .key(key)
+            .send()
+            .await?;
+
+        let data = result.body.collect().await?;
+        Ok(data.into_bytes())
+    }
+
+    async fn list_objects(&self) -> eyre::Result<Vec<String>> {
+        let objects = self
+            .client
+            .list_objects_v2()
+            .bucket(&self.bucket)
+            .send()
+            .await?;
+
+        Ok(objects
+            .contents()
+            .iter()
+            .filter_map(|obj| obj.key().map(String::from))
+            .collect())
+    }
+}
+
+#[derive(Debug, Deserialize)]
+struct CsvIrisRecord {
+    id:         String,
+    left_code:  String,
+    left_mask:  String,
+    right_code: String,
+    right_mask: String,
+}
+
+fn hex_to_bytes(hex: &str, byte_len: usize) -> eyre::Result<Vec<u8>> {
+    if hex.is_empty() {
+        return Ok(vec![]);
+    }
+    let mut bytes = vec![0; byte_len];
+    hex::decode_to_slice(hex, &mut bytes)?;
+    Ok(bytes)
+}
+
+pub async fn last_snapshot_timestamp(store: &impl ObjectStore) -> eyre::Result<i64> {
+    store
+        .list_objects()
```

Comment:
Check this from @wojciechsromek 's PR: https://github.com/worldcoin/iris-mpc-db-exporter/pull/2/files#diff-713f849da8ca4d26c1bf4982ce4d41026a05f017bd78ad3c869c1c4bb51f0b08R63-R66 

I am not entirely sure that `listObectsV2` returns everything - if there are more than 1k rows, the result will be truncated, and the code needs to get a continuation token for it

Reply:
https://docs.aws.amazon.com/cli/latest/reference/s3api/list-objects-v2.html#description